quarkus/tests/integration/src/test-providers/java/org/keycloak/it/resource/realm/TestRealmResource.java

```diff
@@ -17,20 +17,33 @@
 
 package org.keycloak.it.resource.realm;
 
+import org.infinispan.Cache;
+import org.infinispan.commons.configuration.io.ConfigurationWriter;
+import org.infinispan.commons.io.StringBuilderWriter;
+import org.infinispan.configuration.parsing.ParserRegistry;
+import org.jboss.logging.Logger;
+import org.keycloak.connections.infinispan.InfinispanConnectionProvider;
+import org.keycloak.services.resource.RealmResourceProvider;
+
 import jakarta.ws.rs.GET;
 import jakarta.ws.rs.Path;
+import jakarta.ws.rs.PathParam;
 import jakarta.ws.rs.Produces;
 import jakarta.ws.rs.core.MediaType;
 import jakarta.ws.rs.core.Response;
-import org.jboss.logging.Logger;
-import org.keycloak.services.resource.RealmResourceProvider;
 
 /**
  * @author Vaclav Muzikar <vmuzikar@redhat.com>
  */
 public class TestRealmResource implements RealmResourceProvider {
     protected static final Logger logger = Logger.getLogger(TestRealmResource.class);
-    
+
+    final InfinispanConnectionProvider infinispanConnectionProvider;
+
+    public TestRealmResource(InfinispanConnectionProvider infinispanConnectionProvider) {
```

Comment:
To be more future proof, I'd rather suggest passing the whole `KeycloakSession` here instead of a the specific provider.